anchor-service/src/time_event.rs

```diff
@@ -0,0 +1,170 @@
+use anyhow::{anyhow, Result};
+
+use ceramic_event::unvalidated::RawTimeEvent;
+
+use crate::{
+    anchor::{AnchorRequest, TimeEvents},
+    DetachedTimeEvent,
+};
+
+pub fn build_time_events(
+    anchor_requests: &[AnchorRequest],
+    detached_time_event: &DetachedTimeEvent,
+    count: u64,
+) -> Result<TimeEvents> {
+    let events = anchor_requests
+        .iter()
+        .enumerate()
+        .map(|(index, anchor_request)| {
+            let local_path = index_to_path(index.try_into()?, count)?;
+            let remote_path = detached_time_event.path.as_str();
+            Ok(RawTimeEvent::new(
+                anchor_request.id,
+                anchor_request.prev,
+                detached_time_event.proof,
+                format!("{}/{}", remote_path, local_path)
+                    .trim_matches('/')
+                    .to_owned(),
+            ))
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    Ok(TimeEvents { events })
+}
+
+pub fn index_to_path(index: u64, count: u64) -> Result<String> {
+    // we want to find the path to the index in a tree length.
+    // first find the sub-tree then append the path in the sub-tree
+    //
+    // e.g. 5, 6
```

Comment:
I don't quite follow this.  I think the example text below the diagram is saying that the example is with `index` as 10 and `count` as 14?  Can we make that explicit in the comment?  Also, what does the "e.g. 5,6" mean here?